### PR TITLE
added subject filtering at produce function

### PR DESCRIPTION
**Pull request by daniel-davidd**

### CODE REVIEW

File: memphis/memphis.py [hunk 1]
```diff
@@ -545,6 +553,45 @@ def parse_descriptor(self, station_name):
         except Exception as e:
             raise MemphisError(str(e)) from e
 
+    async def start_listen_for_functions_updates(self, station_name, first_functions):
+        #first_functions should contain the dict of the first function of each partition key: partition number, value: first function id
+
+        if station_name in self.functions_updates_subs:
+            self.functions_clients_per_station[station_name] += 1
```
Comment (idanasulinmemphis) [inline memphis/memphis.py:560]: Did you check it? I don't think the we can do += in python

File: memphis/memphis.py [hunk 2]
```diff
@@ -545,6 +553,45 @@ def parse_descriptor(self, station_name):
...
+        sub = await self.broker_manager.subscribe(functions_updates_subject)
+        self.functions_updates_subs[station_name] = sub
+
+        loop = asyncio.get_event_loop()
+        task = loop.create_task(
+            self.get_msg_functions_updates(
+                station_name, self.functions_updates_subs[station_name].messages
+            )
+        )
+        self.functions_tasks[station_name] = task
```
Comment (idanasulinmemphis) [inline memphis/memphis.py:581]: Please cancel these tasks and stop all functions operation where needed, when destroying producers/stations and on connection close, like we are doing in Schemaverse

File: memphis/producer.py [hunk 3]
```diff
@@ -251,6 +260,23 @@ async def destroy(self):
                 if sub is not None:
                     await sub.unsubscribe()
 
+
+            self.connection.functions_clients_per_station[internal_station_name] -= 1
```
Comment (idanasulinmemphis) [inline memphis/producer.py:264]: Please make sure -= is supported in python I don't think so

**Review by idanasulinmemphis — CHANGES_REQUESTED** [review]

File: memphis/memphis.py (hunk 2 above)
Comment (daniel-davidd) [reply memphis/memphis.py:581]: I handle the destruction in producer.py line: 264

**Review by idanasulinmemphis — APPROVED** [review]

**Pull request merged**